packages/manager/apps/key-management-service/src/pages/credential/CreateCredential.page.tsx

```diff
@@ -62,24 +79,39 @@ const CreateCredential = () => {
           headerButton: <KmsGuidesHeader />,
         }}
       >
-        <div className="w-full block">
-          <div className="mb-6">
-            <Notifications />
-          </div>
-          <div className="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-3 gap-4 md:gap-6">
-            <CreateGeneralInformations
-              name={name}
-              setName={setName}
-              validity={validity}
-              setValidity={setValidity}
-              description={description}
-              setDescription={setDescription}
-              csr={csr}
-              setCsr={setCsr}
-            />
+        <IdentityDataProvider>
+          <div className="w-full block">
+            <div className="mb-6">
+              <Notifications />
+            </div>
+            <div className="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-3 gap-4 md:gap-6">
+              {step === 1 && (
+                <CreateGeneralInformations
+                  name={name}
+                  setName={setName}
+                  validity={validity}
+                  setValidity={setValidity}
+                  description={description}
+                  setDescription={setDescription}
+                  csr={csr}
+                  setCsr={setCsr}
+                  nextStep={() => setStep(2)}
+                />
+              )}
+              {step === 2 && (
+                <CreateAddIdentities
+                  identityURNs={identityURNs}
+                  setIdentityURNs={setIdentityURNs}
+                  prevStep={() => {
+                    setStep(1);
+                  }}
+                  nextStep={createCredential}
+                ></CreateAddIdentities>
```

Comment:
```suggestion
                  prevStep={() => setStep(1)}
                  nextStep={createCredential}
                />
```

And I think you can inline directly the createCredential function since it is used only once

Reply:
There is more step coming, i keep this in mind but it miss a step yet, it's mostly used for testing purposes for now while the last step is missing.

Reply:
Well I just suggested a more concise code
The other steps are not placed in one another so it should not be problematic to use a self-closing tag